app/downstream_payload.rb

```diff
@@ -52,27 +44,40 @@ def unpublishing
     edition.unpublishing
   end
 
-  def content_payload
-    Presenters::EditionPresenter.new(
-      edition, draft: draft
-    ).for_content_store(payload_version)
+  def content_presenter
+    Presenters::EditionPresenter.new(edition, draft: draft)
+  end
+
+  def redirect_presenter
+    RedirectPresenter.from_edition(edition)
+  end
+
+  def gone_presenter
+    GonePresenter.from_edition(edition)
   end
 
-  def redirect_payload
-    RedirectPresenter.present(
-      base_path: base_path,
-      publishing_app: edition.publishing_app,
-      public_updated_at: unpublishing.created_at,
-      redirects: unpublishing.redirects,
-    ).merge(payload_version: payload_version)
+  def vanish_presenter
+    VanishPresenter.from_edition(edition)
   end
 
-  def gone_payload
-    GonePresenter.present(
-      base_path: base_path,
-      publishing_app: edition.publishing_app,
-      alternative_path: unpublishing.alternative_path,
-      explanation: unpublishing.explanation,
-    ).merge(payload_version: payload_version)
+  def content_store_presenter
+    return content_presenter unless unpublished?
+
+    case unpublishing.type
+    when "redirect" then redirect_presenter
+    when "gone" then gone_presenter
+    else content_presenter
+    end
+  end
+
+  def message_queue_presenter
+    return content_presenter unless unpublished?
+
+    case unpublishing.type
+    when "redirect" then redirect_presenter
+    when "gone" then gone_presenter
+    when "vanish" then vanish_presenter
+    else content_presenter
```

Comment:
Yes, it will fall through to the content presenter, which is fine for withdrawn since the payload format is the same for a published item.